docs/metrics.md

```diff
@@ -0,0 +1,54 @@
+# GCSFuse Metrics
+GCSFuse supports exporting custom metrics to Google cloud monitoring. 
+Metrics are collected using OpenCensus and exported via Stackdriver exporter.
+
+As of today, GCSFuse exports following metrics related to filesystem and
+gcs calls.
+
+## File system metrics:
+* **fs/ops_count:** Cumulative number of operations processed by file system. It allows
+grouping by op_type to get counts for individual operations. 
+* **fs/ops_error_count:** Cumulative number of errors generated by file system operations.
+Similar to fs/ops_count, this metric can also be grouped by op_type and error_type. 
+* **fs/ops_latency:** Cumulative distribution of file system operation latencies. We 
+can group by op_type.
+
+## GCS metrics
+* **gcs/download_bytes_count:** Cumulative number of bytes downloaded from GCS along
+with read type. Read type specifies sequential or random read.
+* **gcs/read_bytes_count:** Cumulative number of bytes read from GCS objects. This
+is different from download_bytes_count. For eg: we might download x number of
+bytes from GCS but read only <x bytes.
+* **gcs/reader_count:** Cumulative number of GCS object readers opened or closed. We 
+can group the data by IO Method type i.e., opened or closed. 
+* **gcs/request_count:** Cumulative number of GCS requests processed. 
+* **gcs/request_latencies:** Cumulative distribution of the GCS request latencies. 
+* **gcs/read_count:** Specifies the count of gcs reads made along with read type. 
+Read type specifies sequential or random read.
+
+Note: Both request_count and request_latencies allows grouping by gcs method type.
+
+# Usage
+1. We need to set **stackdriver-export-interval** flag to enable exporting metrics to 
```

Comment:
Couple of inputs here
a. Let's create an internal or github issue to record this experimental -> production rollout where we should document the reasons and rationales for posterity.
b. Do you want to club the code changes for this with this PR?

Reply:
Done.

Regarding code changes, we still need to update at two places (fio testing & ml tests) that will done after GCSFuse release because they install a specific version of gcsfuse.